scopes/component/component-filters/component-filters.context.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { ComponentType, createContext, ReactNode, useContext, useState } from 'react';
+import { SlotRegistry } from '@teambit/harmony';
+import { ComponentModel } from '@teambit/component';
+
+export type ComponentFiltersSlot = SlotRegistry<ComponentFilters>;
+
+export type ComponentFilterCriteria<State> = {
+  id: string;
+  render: ComponentType<{ components: ComponentModel[] } & React.HTMLAttributes<HTMLDivElement>>;
+  match: (component: ComponentModel, state: State) => boolean;
+  state: State;
+  order?: number;
+};
+
+export type ComponentFilters = ComponentFilterCriteria<any>[];
+
+export type ComponentFilterContextType = {
+  filters: ComponentFilters;
+  updateFilter: (filter: ComponentFilterCriteria<any>) => void;
+  setFilters: (filters: ComponentFilters) => void;
+  matches: (filters: ComponentFilters, components: ComponentModel[]) => { model: ComponentModel; isHidden?: boolean }[];
+};
+
+export const ComponentFilterContext = createContext<ComponentFilterContextType>({
+  filters: [],
+  setFilters: () => {},
+  updateFilter: () => {},
+  matches: (filters, components) => {
```

Comment:
consider renaming to runFilters()
+ just filter the components, don't return isVisible/isHidden